presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -291,7 +292,7 @@ public Expression rewriteLogicalBinaryExpression(LogicalBinaryExpression node, N
         public Expression rewriteFunctionCall(FunctionCall node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
         {
             // A grouping() is preceded by a GroupId node when the source node is projecting a groupid symbol
-            if (!source.getOutputSymbols().contains(new Symbol("groupid")) && node.getName().toString().equals(GROUPING)) {
+            if (!source.getOutputSymbols().stream().anyMatch(symbol -> symbol.getName().contains(GROUPID_SYMBOL_HINT)) && node.getName().toString().equals(GROUPING)) {
```

Comment:
TL;DR: I don't think it's possible.

The closest query I could come up with that fits your example is the one below:
```
SELECT orderkey,
       custkey,
       sum(agg_price) as outer_sum,
       grouping(orderkey, custkey)
FROM
    (SELECT orderkey,
            custkey,
            sum(totalprice) as agg_price
     FROM orders
     GROUP BY GROUPING SETS ((custkey), (orderkey))
     ORDER BY agg_price ASC
     LIMIT 5) as t
GROUP BY orderkey, custkey;
```

The inner query emits a `groupId` symbol which gets pruned by the final `Project` in that query block since it is not referenced anywhere. If the symbol were carried all the way up then the `GROUPING()` in the outermost layer would not be optimized away to a 0 literal. If you look at the plan for this query the `GROUPING()` does get optimized away which means the `groupId` is not carried forward when it is not 'immediately' used. The scope of the symbol is only within it's own `QuerySpecification`. I hope this addresses your concern, let me know if you have more questions.